dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/sync/CompleteDataSetRegistrationSynchronization.java

```diff
@@ -0,0 +1,96 @@
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.hisp.dhis.dxf2.importsummary.ImportSummary;
+import org.hisp.dhis.dxf2.synch.SynchronizationManager;
+import org.hisp.dhis.setting.SettingKey;
+import org.hisp.dhis.setting.SystemSettingManager;
+import org.hisp.dhis.system.util.Clock;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.client.RestTemplate;
+
+import java.util.Date;
+
+/**
+ * @author David Katuscak
+ */
+public class CompleteDataSetRegistrationSynchronization
+{
+    private static final Log log = LogFactory.getLog( CompleteDataSetRegistrationSynchronization.class );
+
+    private SynchronizationManager synchronizationManager;
+
+    private final SystemSettingManager systemSettingManager;
+
+    private final RestTemplate restTemplate;
+
+    @Autowired
+    public CompleteDataSetRegistrationSynchronization( SynchronizationManager synchronizationManager,
+        SystemSettingManager systemSettingManager, RestTemplate restTemplate )
+    {
+        this.synchronizationManager = synchronizationManager;
+        this.systemSettingManager = systemSettingManager;
+        this.restTemplate = restTemplate;
+    }
+
+
+    public SynchronizationResult syncCompleteDataSetRegistrationData()
+    {
+        if ( !SyncUtils.testServerAvailability( systemSettingManager, restTemplate ).isAvailable() )
+        {
+            return SynchronizationResult.newFailureResultWithMessage( "Complete data set registration synchronization failed. Remote " +
+                "server is unavailable." );
+        }
+        final Clock clock = new Clock( log ).startClock().logTime( "Starting Complete data set registration synchronization job." );
+
+        // ---------------------------------------------------------------------
+        // Set time for last success to start of process to make data saved
+        // subsequently part of next synch process without being ignored
+        // ---------------------------------------------------------------------
+        ImportSummary importSummary;
+        try
+        {
+            importSummary = synchronizationManager.executeDataSetCompletenessPush();
+            if ( SyncUtils.checkSummaryStatus( importSummary, SyncEndpoint.COMPLETE_DATA_SET_REGISTRATIONS ) )
+            {
+                clock.logTime( "SUCCESS! Complete data set registration synchronization job is done. It took" );
```

Comment:
This way of logging and the message seems a bit strange. Since you both log and return same message, maybe make it a string first and then use the string to log and return?

Reply:
In the returned object, it is not really needed as currently the success is not propagated further. Just feel correct to have it there ...